tensorflow/c/kernels_test.cc

```diff
@@ -589,50 +589,77 @@ TEST(TestKernel, DeleteKernelBuilderIsOkOnNull) {
   TF_DeleteKernelBuilder(nullptr);
 }
 
-TEST(TestKernel, TestTypeConstraint) {
-  const char* node_name = "SomeNodeName";
-  const char* op_name = "TypeOp";
-  const char* device_name = "FakeDeviceName1";
-
-  REGISTER_OP(op_name)
-      .Input("input1: double")
-      .Input("input2: uint8")
-      .Output("output1: uint8")
-      .Attr("T: type");
-
-  TF_KernelBuilder* builder = TF_NewKernelBuilder(
-      op_name, device_name, &MyCreateFunc, &MyComputeFunc, &MyDeleteFunc);
-  TF_Status* status = TF_NewStatus();
-  TF_KernelBuilder_TypeConstraint(builder, "T", TF_DataType::TF_INT32, status);
-  EXPECT_EQ(TF_OK, TF_GetCode(status));
-  TF_RegisterKernelBuilder(node_name, builder, status);
-  EXPECT_EQ(TF_OK, TF_GetCode(status));
-
-  TF_Buffer* buf = TF_GetRegisteredKernelsForOp(op_name, status);
-  EXPECT_EQ(TF_OK, TF_GetCode(status));
-  KernelList list;
-  list.ParseFromArray(buf->data, buf->length);
+std::string expected_string(const char* type) {
   const auto expected_str = R"str(kernel {
-  op: "TypeOp"
+  op: "TypeOp)str" + std::string(type) +
+                            std::string("\"") +
+                            R"str(
   device_type: "FakeDeviceName1"
   constraint {
     name: "T"
     allowed_values {
       list {
-        type: DT_INT32
+        type: )str" + std::string(type) +
+                            R"##(
```

Comment:
May be use `StrFormat` instead to make it easier to read. Example: https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/stream_executor/gpu/redzone_allocator.cc;l=69;drc=4648d99eaecddd47a1fd5209a620635733726492

Reply:
Done. And raw string is necessary here for comparing with protobuf generated string.